### PR TITLE
TASK-3-create-template

### PR DESCRIPTION
**Pull request by bombenko**

### CODE REVIEW

File: src/components/dump/Task.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export class Task {
+  constructor(title, description) {
+    this.title = title;
+    this.description = description;
+    this.createTime = new Date();
+    this.isCompleted = false;
+  }
+  setComplited() {
```
Comment (grachpower) [inline src/components/dump/Task.js]: 'Complited' :D

File: src/components/smart/Card.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { Task } from '../dump/Task';
+
+export class Card {
+  constructor(name, taskArr = []) {
```
Comment (grachpower) [inline src/components/smart/Card.js]: prevent using default values

File: src/components/smart/Card.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    this.name = name;
+    this.taskArr = taskArr;
+  }
+  createTask(title, description) {
+    const task = new Task(title, description);
+    this.taskArr.push(task);
+  }
+  findTask(task) {
+    for (let i = 0; i < this.taskArr.length; i++) {
```
Comment (grachpower) [inline src/components/smart/Card.js]: https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/find

File: src/components/smart/Card.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      if (this.taskArr[i] === task) {
+        return i;
+      }
+    }
+    return -1;
+  }
+  deleteTask(task) {
+    const i = this.findTask(task);
+    if (i !== -1) {
```
Comment (grachpower) [inline src/components/smart/Card.js]: too complex

File: src/components/smart/Card.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      this.taskArr.splice(i, 1);
+      return true;
+    }
+    return false;
+  }
+  editTask(task, title, description) {
```
Comment (grachpower) [inline src/components/smart/Card.js]: too complex

File: src/css/style.css [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+	color: #222;
+	-webkit-font-smoothing: antialiased;
+	-moz-osx-font-smoothing: grayscale;
+	-webkit-overflow-scrolling: touch;
+	margin: 0;
+  background-color: #50BCC5;
+}
+
+p{
+  margin-bottom: 0;
```
Comment (grachpower) [inline src/css/style.css]: вот тут есть отступ

File: src/css/style.css [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+  margin-top: 0;
+}
+
+.title{
+    margin-top: 0;
+    margin-bottom: 0;
```
Comment (grachpower) [inline src/css/style.css]: а тут другой отступ

File: src/index.js [hunk 8]
```diff
@@ -1,11 +1,109 @@
-import { join } from 'lodash';
+import { Card } from './components/smart/Card';
+import './css/style.css';
 
-function component() {
-  const element = document.createElement('div');
+const mn = new Card('Monday');
```
Comment (grachpower) [inline src/index.js]: use full name for contacts such as monday/tuesday

File: src/index.js [hunk 9]
```diff
@@ -1,11 +1,109 @@
...
+mn.createTask('Prif', 'Che del');
```
[the remark this answers is not in the document]
Comment (grachpower) [reply src/index.js]: same as above

File: src/index.js [hunk 10]
```diff
@@ -1,11 +1,109 @@
...
+mn.createTask('Vilkoi', 'V glaz');
```
[the remark this answers is not in the document]
Comment (grachpower) [reply src/index.js]: same as above

File: src/index.js [hunk 11]
```diff
@@ -1,11 +1,109 @@
...
+mn.createTask('V popu', 'Raz');
 
-  element.innerHTML = join(['Hello', 'webpack'], ' ');
+const tu = new Card('Tuesday');
```
Comment (grachpower) [inline src/index.js]: same as above

File: src/index.js [hunk 12]
```diff
@@ -1,11 +1,109 @@
...
+tu.createTask('Prif', 'Che del');
+tu.createTask('Vilkoi', 'V glaz');
+tu.createTask('V popu', 'Raz');
 
-  return element;
+const we = new Card('Wednesday');
```
Comment (grachpower) [inline src/index.js]: see comment above

File: src/index.js [hunk 13]
```diff
@@ -1,11 +1,109 @@
...
+const su = new Card('Sunday');
+su.createTask('Vilkoi', 'V glaz');
+su.createTask('V popu', 'Raz');
+
+const days = [mn, tu, we, th, fr, sa, su];
+
+/**
+ *
+ * @param {Task} task
+ * @param {num} index
```
Comment (grachpower) [inline src/index.js]: we have primitive type number

File: webpack.config.js [hunk 14]
```diff
@@ -18,12 +18,11 @@ module.exports = {
   module: {
     rules: [
       {
-        test: /\.less$/,
```
Comment (grachpower) [inline webpack.config.js]: use less with constants

File: src/components/dump/Task.js (hunk 1 above)
Comment (bombenko) [reply src/components/dump/Task.js]: Как слышится, так и пишется

File: src/index.js [hunk 15]
```diff
@@ -1,11 +1,109 @@
...
+    </div>
+  `;
+}
+
+/**
+ *
+ * @param {Array} cardList
+ */
+function getCardListTemplate(cardList) {
+  return (cardList.map((card) => getCardTemplate(card))).join('');
```
Comment (grachpower) [inline src/index.js]: return cardList.map((card) => getCardTemplate(card)).join('');

File: src/index.js [hunk 16]
```diff
@@ -1,11 +1,109 @@
...
+      <div class="button-wrapper">
+        <button type="button" class="add-button">Add element...</button>
+      </div>   
+    </div>
+  `;
+}
+
+/**
+ *
+ * @param {Array} cardList
```
Comment (grachpower) [inline src/index.js]: @param {Array<Card>} cardList

File: src/components/smart/Card.js (hunk 5 above)
Comment (AlenaMordvinceva) [reply src/components/smart/Card.js]: Uncle Bob allows 3 parameters

File: src/index.js [hunk 17]
```diff
@@ -1,11 +1,116 @@
...
+        <div class="main-container">    
+          ${getCardListTemplate(days)}
+        </div> 
+    </main>
+</body>
+</html>
+`;
+
+createDefaultTask();
+document.body.innerHTML = html;
```
Comment (grachpower) [inline src/index.js]: внутри body будет конструкция 
`+<!DOCTYPE html>
 +<html>
 +<head>
 +    <title>ToDoList</title>
 +    <meta name="viewport" content="width=device-width,init-scale=1">
 +</head>
 +<body>
 +    <div class="header">
 +        <div class="main-container">
 +            <h1 class="header__title">Get it done</h1>
 +        </div>
 +    </div>
 +    <main class="content">
 +        <div class="main-container">    
 +          ${getCardListTemplate(days)}
 +        </div> 
 +    </main>
 +</body>
 +</html>`

File: src/components/smart/Card.js [hunk 18]
```diff
@@ -34,15 +34,18 @@ export class Card {
    * @param {Task} task
    * @param {string} title
    * @param {string} description
+   *
    * @returns {boolean}
    */
   editTask(task, title, description) {
     const i = this.taskList.indexOf(elem => elem === task);
     if (i !== -1) {
```
Comment (AlenaMordvinceva) [inline src/components/smart/Card.js]: empty row before if

**Review by AlenaMordvinceva — APPROVED** [review]

File: src/components/smart/Card.js (hunk 18 above)
Comment (AlenaMordvinceva) [reply src/components/smart/Card.js]: to clear code styles with if cases
// RIGHT
 public olololo(): void {
    if (true !== false) {
      if (false === false) {
        const kek: boolean = this.startToDoSmthg();

        if (kek) {
        .....
        }
      }
    }
  }


// WRONG check comments why wrong
 public olololo(): void {
 
if (true !== false) {      // it is the first statement in this lexical env, so there is no empty row
     
if (false === false) {     //same here
        const kek: boolean = this.startToDoSmthg();
        if (kek) {      // within one lexical env empty row if 'if' statement is going after smthg
        .....
        }
      }
    }
  }

**Review by grachpower — APPROVED** [review]

**Pull request merged**